src/scales/scale.time.js

```diff
@@ -434,28 +477,14 @@ function getLabelBounds(scale) {
 	return {min, max};
 }
 
-/**
- * Return subset of `timestamps` between `min` and `max`.
- * Timestamps are assumend to be in sorted order.
- * @param {int[]} timestamps - array of timestamps
- * @param {int} min - min value (timestamp)
- * @param {int} max - max value (timestamp)
- */
-function filterBetween(timestamps, min, max) {
-	let start = 0;
-	let end = timestamps.length - 1;
-
-	while (start < end && timestamps[start] < min) {
-		start++;
-	}
-	while (end > start && timestamps[end] > max) {
-		end--;
-	}
-	end++; // slice does not include last element
-
-	return start > 0 || end < timestamps.length
-		? timestamps.slice(start, end)
-		: timestamps;
+function generateTicks(scale) {
+	const options = scale.options;
+	const timeOpts = options.time;
+	const source = options.ticks.source;
+	const maxCapacity = options.ticks.maxTicksLimit || Math.max(Math.ceil(scale._length / scale._resolveTickFontOptions(0).lineHeight / 2), 1) || 1;
+	const units = determineUnits(scale, maxCapacity);
+	const stepSize = resolve([timeOpts.stepSize, timeOpts.unitStepSize, determineStepSize(scale, units[0], maxCapacity)]);
```

Comment:
I agree that having a tick for every data point is too much and we should always generate ticks based on the unit.

I think using a `stepSize` of `1` and relying on `autoSkip` will work a better than `determineStepSize` though. `maxCapacity` relies on `_length`, which won't be correct as it's updated during `fit`. There's a couple problems with getting it wrong here. One is that getting it a little wrong will compound when `autoSkip` is turned on. E.g. imagine 18 ticks can fit and `capacity` here says 19 can fit. Then when you `autoSkip` you skip every other one and end up with only 10. If we'd just handled it in `autoSkip` we might have been able to fit much closer to 18. Another issue is that `stepSize` might step over major ticks giving the wrong result. Finally it's just less code duplication and room for error. Why have two places to calculate how many ticks can fit?

Reply:
Thanks for taking a look!

I some adjustments.
 - if `autoSkip` is enabled, generate enough ticks (up to 1000, stepSize=1).
 - if not, check if rotation is enabled and generate more ticks if it is.
 - align to weeks and quarters (removed the `common` as it is not universal)

`_length` does not seem to change in simple chart, with only 2 scales.

I'm more worried about the generation of ticks than the "how many can fit".
I'll add a plunker link to description to show more of the differences.

Reply:
> `_length` does not seem to change in simple chart, with only 2 scales.

I'm used to seeing it change. I just tested the financial sample right now and the x-axis `_length` changed from 948.49 to 940.03

> align to weeks and quarters (removed the common as it is not universal)

I'm pretty sure we'd gotten bug reports about this. It's pretty weird to see "week 26 2018", "week 27 2018", "week 28 2018", etc. It's a pretty meaningless way to align ticks. No one thinks about dates that way. I'd rather remove `week` and `quarter` support entirely than have them possibly automatically chosen. I think we had a good compromise before where you could manually choose them, but they wouldn't be chosen automatically

Reply:
> I'm pretty sure we'd gotten bug reports about this. It's pretty weird to see "week 26 2018", "week 27 2018", "week 28 2018", etc. It's a pretty meaningless way to align ticks. No one thinks about dates that way. I'd rather remove week and quarter support entirely than have them possibly automatically chosen. I think we had a good compromise before where you could manually choose them, but they wouldn't be chosen automatically

Are you thinking more of the formatting than alignment? All calendars I've seen are aligned to weeks, having those numbered or not.

I was actually thinking of letting the available units be configured. Should cover all cases then.